mmv1/third_party/terraform/acctest/tgc_utils.go

```diff
@@ -1,64 +1,307 @@
 package acctest
 
 import (
+	"encoding/base64"
+	"encoding/json"
 	"fmt"
 	"log"
 	"regexp"
+	"runtime"
 	"strings"
 
 	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
 	"github.com/hashicorp/terraform-plugin-testing/terraform"
 )
 
+type ResourceMetadata struct {
+	CaiAssetName    string         `json:"cai_asset_name"`
+	ResourceType    string         `json:"resource_type"`
+	ResourceAddress string         `json:"resource_address"`
+	ImportMetadata  ImportMetadata `json:"import_metadata,omitempty"`
+	Service         string         `json:"service"`
+}
+
+type ImportMetadata struct {
+	Id            string   `json:"id,omitempty"`
+	IgnoredFields []string `json:"ignored_fields,omitempty"`
+}
+
+type TgcMetadataPayload struct {
+	TestName         string                      `json:"test_name"`
+	RawConfig        string                      `json:"raw_config"`
+	ResourceMetadata map[string]ResourceMetadata `json:"resource_metadata"`
+	PrimaryResource  string                      `json:"primary_resource"`
+}
+
 // Hardcode the Terraform resource name -> API service name mapping temporarily.
 // TODO: [tgc] read the mapping from the resource metadata files.
 var ApiServiceNames = map[string]string{
 	"google_compute_instance": "compute.googleapis.com",
 	"google_project":          "cloudresourcemanager.googleapis.com",
 }
 
-// Gets the test metadata for tgc:
-//   - test config
-//   - cai asset name
-//     For example: //compute.googleapis.com/projects/ci-test-188019/zones/us-central1-a/instances/tf-test-mi3fqaucf8
-func GetTestMetadataForTgc(service, address, rawConfig string) resource.TestCheckFunc {
+// encodeToBase64JSON converts a struct to base64-encoded JSON
+func encodeToBase64JSON(data interface{}) (string, error) {
+	jsonData, err := json.Marshal(data)
+	if err != nil {
+		return "", fmt.Errorf("error marshalling data to JSON: %v", err)
+	}
+
+	return base64.StdEncoding.EncodeToString(jsonData), nil
+}
+
+// CollectAllTgcMetadata collects metadata for all resources in a test step
+func CollectAllTgcMetadata(config string, resourceMetadata map[string]ResourceMetadata) resource.TestCheckFunc {
 	return func(s *terraform.State) error {
-		splits := strings.Split(address, ".")
-		if splits == nil || len(splits) < 2 {
-			return fmt.Errorf("The resource address %s is invalid.", address)
+		// Get the name of the test that invoked this function
+		testName := getCallerInfo()
```

Comment:
t.Name can be used to get the test name. Or I missing something.

Reply:
t is not in scope of the vcr functions

Reply:
`t` is the parameter of the function [func VcrTest(t *testing.T, c resource.TestCase) {](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/acctest/vcr_utils.go#L142), where `extendWithTGCData(c)` is called.

Reply:
This is true, I'll make the change